src/Resolver.js

```diff
@@ -276,16 +276,14 @@ class Resolver {
     // libraries like d3.js specifies node.js specific files in the "main" which breaks the build
     // we use the "module" or "browser" field to get the full dependency tree if available.
     // If this is a linked module with a `source` field, use that as the entry point.
-    let main = [pkg.source, pkg.module, browser, pkg.main].find(
-      entry => typeof entry === 'string'
-    );
-
-    // Default to index file if no main field find
-    if (!main || main === '.' || main === './') {
-      main = 'index';
+    for (let source of [pkg.source, pkg.module, browser, pkg.main, 'index']) {
+      if (
+        typeof source === 'string' &&
+        (await fs.exists(path.resolve(pkg.pkgdir, source)))
```

Comment:
Hi!

According to node.js docs `fs.exists` method [is deprecated](https://nodejs.org/api/fs.html#fs_fs_exists_path_callback). In order to check if files exists we can use [`fs.access`](https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback) method or sync version [`fs.accessSync`](https://nodejs.org/api/fs.html#fs_fs_accesssync_path_mode).